net/url/url.h

```diff
@@ -115,20 +118,25 @@ struct URL {
     }
   }
 
-  URL(const std::string& url, const URL& previous)
-      : URL(url, previous.protocol, previous.host, previous.port) {}
+  URLWithoutParameters(const std::string& url, const URLWithoutParameters& previous)
+      : URLWithoutParameters(url, previous.protocol, previous.host, previous.port) {}
 
   std::string ComposeURL() const {
-    std::ostringstream os;
-    if (!protocol.empty()) {
-      os << protocol << "://";
-    }
-    os << host;
-    if (port != DefaultPortForProtocol(protocol)) {
-      os << ':' << port;
+    if (!host.empty()) {
+      std::ostringstream os;
+      if (!protocol.empty()) {
+        os << protocol << "://";
+      }
+      os << host;
```

Comment:
wanna be cool? you do not parse user:password part in scheme://user:password@host:port/path?query